lib/app_logs.rb

```diff
@@ -1,30 +1,26 @@
 # frozen_string_literal: true
 
+require 'fileutils'
 require 'pathname'
 
 module AppLogs
   class << self
     def registered(app)
-      @env = app.settings.environment
-      app.set :access_log, access_log!
-      app.set :error_log, error_log!
+      environment = app.settings.environment
+      app.set :access_log, open_log_file!(environment, 'access')
+      app.set :error_log, open_log_file!(environment, 'error')
 
       app.enable :logging
       app.use Rack::CommonLogger, app.settings.access_log
 
       app.before { env['rack.errors'] = app.settings.error_log }
     end
 
-    def access_log!
-      log_path('access').open('a+').tap { |log| log.sync = true }
-    end
-
-    def error_log!
-      log_path('error').open('a+').tap { |log| log.sync = true }
-    end
+    private
 
-    def log_path(type)
-      Pathname(__dir__).expand_path.join('../log', "#{@env}_#{type}.log")
+    def open_log_file!(env, name)
+      log_dir = Pathname(__dir__).join('../log').expand_path.tap { |dir| FileUtils.mkdir_p(dir) }
```

Comment:
Since it's a `Pathname...`

```suggestion
      log_dir = Pathname(__dir__).join('../log').expand_path.tap { |dir| dir.mkpath }
```